transformation/transformation.go

```diff
@@ -0,0 +1,703 @@
+			return (a != 0.0) != (b != 0.0), nil
+		}
+	case string:
+		if b, ok := b.(string); ok {
+			return (a != "") != (b != ""), nil
+		}
+	case []byte:
+		if b, ok := b.([]byte); ok {
+			return (len(a) != 0) != (len(b) != 0), nil
+		}
+	case civil.Date:
+		if b, ok := b.(civil.Date); ok {
+			return (a != civil.Date{}) != (b != civil.Date{}), nil
+		}
+	case time.Time:
+		if b, ok := b.(time.Time); ok {
+			return (!a.IsZero()) != (!b.IsZero()), nil
+		}
+	}
+	return false, fmt.Errorf("unsupported type for logical operation: %T", a)
+}
+
+// Logical NOT operator
+func logicalNot(a interface{}) (bool, error) {
+	switch a := a.(type) {
+	case bool:
+		return !a, nil
+	case int64:
+		return a == 0, nil
+	case float64:
+		return a == 0.0, nil
+	case string:
+		return a == "", nil
+	case []byte:
+		return len(a) == 0, nil
+	case civil.Date:
+		return a == civil.Date{}, nil
+	case time.Time:
+		return a.IsZero(), nil
+	}
+	return false, fmt.Errorf("unsupported type for logical operation: %T", a)
+}
+
+func applyCompare(firstInput, operator, secondInput inputValue) (interface{}, error) {
+	switch operator.value {
+	case EqualToOperator:
+		return compareEqual(firstInput, secondInput)
+	case GreaterThanOperator:
+		return compareGreaterThan(firstInput, secondInput)
+	case LesserThanOperator:
+		return compareLessThan(firstInput, secondInput)
+	}
+	return nil, fmt.Errorf("unsupported comparison operation: %s", operator.value)
+}
+
+func compareEqual(firstInput, secondInput inputValue) (bool, error) {
+	if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Int64 {
+		return firstInput.value.(int64) == secondInput.value.(int64), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Float64 {
+		return firstInput.value.(float64) == secondInput.value.(float64), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Bool {
+		return firstInput.value.(bool) == secondInput.value.(bool), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Bytes {
+		return string(firstInput.value.([]byte)) == string(secondInput.value.([]byte)), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Date {
+		return firstInput.value.(civil.Date) == secondInput.value.(civil.Date), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Timestamp {
+		return firstInput.value.(time.Time).Equal(secondInput.value.(time.Time)), nil
+	} else if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.String {
+		return firstInput.value == secondInput.value, nil
+	}
+	return false, fmt.Errorf("unsupported data types for comparison: %T, %T", firstInput.value, secondInput.value)
+}
+
+func compareGreaterThan(firstInput, secondInput inputValue) (bool, error) {
+	if firstInput.dataType == secondInput.dataType && firstInput.dataType == ddl.Int64 {
```

Comment:
this would need to be a number based comparison, not type based. 
Eg:
10 > 9.5